app/routes/messages.rb

```diff
@@ -0,0 +1,24 @@
+Mumukit::Platform.map_organization_routes!(self) do
+  post '/courses/:course/messages' do
+    authorize! :teacher
+    message = json_body[:message]
+    message[:sender] = current_user.uid
+    assignment = Assignment.find_by!(with_organization_and_course 'exercise.eid': json_body[:exercise_id], 'student.uid': json_body[:uid])
+    assignment.add_message! message, json_body[:submission_id]
+    assignment.notify_message! message, json_body[:submission_id]
+    {status: :created, message: Message.new(message)}
+  end
+  get '/courses/:course/guides/:organization/:repository/:uid/:exercise_id/messages' do
```

Comment:
What is the `uid` for? For the student? What is the `exercise_id?`

I would improve this route so that it is more clear in intention: 

```
/courses/:course/guides/:organization/:repository/student/:uid/exercise/:exercise_id/messages' do
```

Or event better: 

```
/courses/:course/exercise/:organization/:repository/:exercise_id/student/:uid/messages' do
```

Becasue: 

* a course is identified by a course `name`
* an exercise is identified by `organization`, `repo` and exercise - bibliotheca - `id`
* a student is identified by an `uid`